File: untold_comparison_programs/05_file_operations/python/main.py
# File Operations
with open("data.txt", "w") as f:
    f.write("Hello from Python!")
print("File written!")

with open("data.txt", "r") as f:
    content = f.read()
print(f"Content: {content}")

import os
exists = os.path.exists("data.txt")
if exists:
    print("data.txt exists!")

os.remove("data.txt")
print("File deleted!")

File: untold_comparison_programs/05_file_operations/go/main.go
package main

import (
    "fmt"
    "os"
    "io/ioutil"
)

func main() {
    err := ioutil.WriteFile("data.txt", []byte("Hello from Go!"), 0644)
    if err != nil {
        fmt.Println("Error:", err)
        return
    }
    fmt.Println("File written!")

    content, err := ioutil.ReadFile("data.txt")
    if err != nil {
        fmt.Println("Error:", err)
        return
    }
    fmt.Printf("Content: %s\n", content)

    _, err = os.Stat("data.txt")
    if err == nil {
        fmt.Println("data.txt exists!")
    }

    err = os.Remove("data.txt")
    if err != nil {
        fmt.Println("Error:", err)
        return
    }
    fmt.Println("File deleted!")
}